File: txtscrio/centered_frame.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void centered_frame (uint8_t *xstart, uint8_t *ystart,
                     uint8_t hsize, uint8_t vsize, const String *name,
                     uint8_t attr1, uint8_t attr2, const String *border)
{
  *xstart = (work_MaxCol - hsize) / 2;
  *ystart = ((work_MaxLn - vsize) / 2) + ((work_MaxLn - vsize) % 2);

  Frame (centered_frame_vdest, *xstart, *ystart, *xstart + hsize, *ystart + vsize,
         attr1, name, attr2, border);
}

File: txtscrio/pas/centered_frame.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

// Saved from "adt2unit.pas"

procedure centered_frame(var xstart,ystart: Byte; hsize,vsize: Byte;
                         name: String; atr1,atr2: Byte; border: String);
begin
  xstart := (work_MaxCol-hsize) DIV 2;
  ystart := (work_MaxLn -vsize) DIV 2+(work_MaxLn-vsize) MOD 2;

  Frame(centered_frame_vdest,xstart,ystart,xstart+hsize,ystart+vsize,
        atr1,name,atr2,border);
end;
